### PR TITLE
Store mempool to mempool.dat on shutdown

### PR DESCRIPTION
**Pull request by ptschip**
This PR works as is, I just need to tidy up the commits a little.

### CODE REVIEW

File: src/validation/validation.cpp [hunk 1]
```diff
@@ -2875,9 +2875,7 @@ bool DisconnectTip(CValidationState &state, const Consensus::Params &consensusPa
             {
                 CTxInputData txd;
                 txd.tx = ptx;
-                txd.nodeId = -1;
                 txd.nodeName = "rollback";
-                txd.whitelisted = false;
```
Comment (gandrewstone) [inline src/validation/validation.cpp:2880]: CTxInputData does not have an initializer, so every field must be set.  Probably these were accidentally removed during a merge since they seem unrelated to the topic of this PR.

File: src/txadmission.cpp [hunk 2]
```diff
@@ -1045,7 +1045,6 @@ void ProcessOrphans(std::vector<uint256> &vWorkQueue)
                     txd.tx = orphanTx;
                     txd.nodeId = fromPeer;
                     txd.nodeName = "orphan";
-                    txd.whitelisted = false;
```
Comment (gandrewstone) [inline src/txadmission.cpp:1048]: must be initialized...

File: src/rpc/rawtransaction.cpp [hunk 3]
```diff
@@ -1042,9 +1042,7 @@ UniValue enqueuerawtransaction(const UniValue &params, bool fHelp)
 
     CTxInputData txd;
     txd.tx = MakeTransactionRef(std::move(tx));
-    txd.nodeId = -1;
     txd.nodeName = "rpc";
-    txd.whitelisted = false;
     EnqueueTxForAdmission(txd);
```
Comment (gandrewstone) [inline src/rpc/rawtransaction.cpp:1046]: ditto (probable accidental merge removal)

File: src/validation/validation.cpp (hunk 1 above)
Comment (ptschip) [reply src/validation/validation.cpp:2880]: there was a constructor added to this commit:   CTxInputData() : nodeId(-1), whitelisted(false), nodeName("none") {}
